Add CI and dependabot

double-beep opened the pull request.

codygray commented:
@double-beep I recently made a bunch of changes to the Java code, refactoring and adding new features, before seeing or evaluating your pull request. Since you're more familiar with the pull request's contents than I am, would you please review the conflicts and see if there's anything that needs to be changed?

I *think* you did not touch any of the Java code files, which is pretty much all that I touched (save for updating the version number in the Maven configuration), so there should not be any *real* merge conflicts, but I would appreciate a second pair of eyes from the author of this pull request.

Thanks! I'll happily merge this in once you review and update as required.

double-beep committed:
Add dependabot

double-beep committed:
Remove unused imports and order them

No functional change

double-beep committed:
Indentation/redundant parentheses/+ on a new line

No functional change

double-beep committed:
Remove unused variables

double-beep committed:
Change order of instance vars

double-beep committed:
Indentation, created Map with the commands

This is temporary in the sense that it will not persist across an unexpected server reboot -> changes will not persist across a reboot

double-beep committed:
Add checkstyle and CI

double-beep committed:
Test on Java 11+

double-beep committed:
Use LOGGER.*

codygray commented:
I've cherry-picked and merged the parts of this changeset that I agree with.

 - I prefer the notation `Type var[]` over `Type[] var` for declarations. Maybe this is my C++ bias, but I think the former is clearer and more accurate.
 - Binary comparison operators should be grouped with parentheses in `if` statements, as they logically resolve to a single result. And, more generally, parentheses should be used everywhere necessary to avoid having to reason about operator precedence and/or associativity rules, even at the risk of some redundancy. Code is written only once, but read many times. It never hurts to have an extra pair of parentheses in there. (Except truly redundant ones, wrapped multiple times around the exact same construct.)
 - Private members should always go at the end of a class implementation because they do not comprise part of the class's consumable interface and they are never what someone who opens up a class's code file is looking for.
 - Removing the unused blacklist URL variables is completely reasonable, but I plan to use them in the future, so I've intentionally left them uncommented.
 - The change to a map is not bad, but not something I want to go to at this time, as it seems to offer no tangible benefits over a string literal. If the map was also used to dispatch/handle messages, then I'd be all for it, but as it stands now, it just introduces unneeded complexity and possibly reduces performance at run-time for what should be a very simple operation.
